app/uk/gov/hmrc/cdsreimbursementclaim/controllers/DeclarationInfoController.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaim.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.JsValue
+import play.api.mvc.{Action, ControllerComponents}
+import uk.gov.hmrc.cdsreimbursementclaim.services.DeclarationInfoService
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging.LoggerOps
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.concurrent.ExecutionContext
+
+@Singleton()
+class DeclarationInfoController @Inject()(declarationInfoService: DeclarationInfoService, cc: ControllerComponents)(
+  implicit ec: ExecutionContext) extends BackendController(cc) with Logging {
+
+  def getDeclarationInfo(): Action[JsValue] = Action.async(parse.json) { implicit request =>
+    declarationInfoService
```

Comment:
We need to add some validation here to make sure we have received a valid request. According to the spec I am looking at (v1.3.0) the `requestCommon` fields are mandatory. Therefore if they are missing then we need to return a BadRequest.

Reply:
No.
We have to do the same in the frontend anyway.
So it would just lead to code duplication

Reply:
My PR comment above yours is not longer valid. However, since we are changing the endpoint to accept a declaration id it would be good to have a query path binder that validates that a correct declaration id is being passed. That will future-proof it in case other teams want to hit this end point (has happened in the past).

Reply:
Already Done